MdePkg/Include/IndustryStandard/IpmiNetFnGroupExtension.h

```diff
@@ -17,4 +20,67 @@
 // All Group Extension commands and their structure definitions to follow here
 //
 
+///
+/// Constants and structure definitions for "SBMR Appendix F Boot Progress Code
+/// reporting" command to follow here
+///
+
+//
+// Definitions for send progress code command
+//
+#define IPMI_GROUP_EXTENSION_BOOT_PROGRESS_CODE_SEND  0x02
+
+//
+// Definitions for get progress code command
+//
+#define IPMI_GROUP_EXTENSION_BOOT_PROGRESS_CODE_GET  0x03
+
+//
+// Definitions for send and get progress code command response
+//
+#define IPMI_GROUP_EXTENSION_BOOT_PROGRESS_CODE_COMPLETED_NORMALLY    0x00
+#define IPMI_GROUP_EXTENSION_BOOT_PROGRESS_CODE_COMPLETED_WITH_ERROR  0x80
+#define IPMI_GROUP_EXTENSION_BOOT_PROGRESS_CODE_DEFINING_BODY         0xAE
+
+//
+// Structure for the format of the boot progress code data
+//
+typedef struct {
+  EFI_STATUS_CODE_TYPE     CodeType;
+  EFI_STATUS_CODE_VALUE    CodeValue;
+  UINT32                   Instance;

```

Comment:
Instance should be UINT8 ?

Reply:
I think it should be spec issue. In the details column of the SBMR specication, it mentions that the Instance matches the Instance parameter of the ReportStatusCode() PEI service [1]. I reported this issue internally, hence, I think we should implement it correctly first.

[1] https://uefi.org/specs/PI/1.8A/V1_Services_PEI.html#reportstatuscode

Reply:
I think this change should be defined in MdePkg\Include\Pi\PiStatusCode.h. If there is the issue in PI spec, the spec change should be raised in PIWG group first.

Reply:
yes, @changab. We will file an ECR to review and discuss this in the SBMR forum. It may take some time. To move forward, should I update it to UINT8 to align with the current SBMR specification? We can then make the necessary corrections once we have a fix for the specification.

Reply:
I think that is fine to update this field to UINT8 as the entire package is sent to BMC that follows the SBMR spec. Using UINT32 may cause the transfer error as the total package size would be considered as an invalid IPMI package. at BMC end. However, could you please add the spec reference in the comment above this structure? Point out how this structure is defined.  I would suggest you send the ECR to just update the description of "Instance" in SBMC spec.  Don't mention the "instance" is compliant with PEI/DXE ReporStatusCode. With this, we can reduce the backward compatibility issue on BMC side and we don't have to update this file again.

Reply:
> However, could you please add the spec reference in the comment above this structure? Point out how this structure is defined.

Sure, I will add it.

> I would suggest you send the ECR to just update the description of "Instance" in SBMC spec. Don't mention the "instance" is compliant with PEI/DXE ReporStatusCode. With this, we can reduce the backward compatibility issue on BMC side and we don't have to update this file again.

That sounds great. Thank you for the suggestion. I've taken note of it.

Reply:
I have updated the instance field to UINT8. Resolving this thread.

Reply:
@abdattar Hi Abdul, do you have any comments on this?

Reply:
Looks good to me.